modules/compression/src/lib/zstd-compression.ts

```diff
@@ -50,4 +52,18 @@ export class ZstdCompression extends Compression {
     const inputArray = new Uint8Array(input);
     return simpleZstd.decompress(inputArray).buffer;
   }
+
+  async decompress(input: ArrayBuffer, size?: number): Promise<ArrayBuffer> {
+    await this.preload();
+    const simpleZstd = new zstd.Streaming();
+    const inputArray = new Uint8Array(input);
+
+    const chunks: ArrayBuffer[] = [];
+    for (let i = 0; i <= inputArray.length; i += CHUNK_SIZE) {
+      chunks.push(input.slice(i, i + CHUNK_SIZE));
```

Comment:
Is there any way of just passing in a typed array view of the 1MB data, rather than creating new 1MB ArrayBuffers with `.slice()`.

Reply:
Done